### DISCUSSION TITLE
Add option to resize text automatically, if it exceeds page length

### DISCUSSION THREAD

**Discussion by JPustkuchen** [slidevjs/slidev#1108] [Ideas]
**Is your feature request related to a problem? Please describe.**
Longer texts or lists easily exceed the presentation page height. When writing the Markdown content, you don't want to care for this manually. Just write down the important things and be sure it fits the slide.

**Describe the solution you'd like**
If a page content exceeds the page size (height) it would be super helpful to have a global or per page setting like

```
---
autosize: true
---
```
which then shrinks the contents like `<Transform>`.

When used in the slides.md head, it should be global. When used as page setting, just for the page.

Additionally, one could think about an `<Autosize>` element, if it makes sense technically.

I know, technically, this might be hard! But it would be really great and honestly the last missing big thing for Slidev I'm missing!
Whao, such a great tool! Would be nice if we could figure out a solution.

**Describe alternatives you've considered**
Currently, the workaround is to check the slides manually and use the `<Transform scale="0.x">` wrapper, but that's less convenient

**Comment by JPustkuchen** [comment]
Another alternative teminology could be `autofit: true`

**Comment by marshallfWCCI** [comment]
If you're interested in fitting to a page, you will probably find https://github.com/slidevjs/slidev/issues/1075 useful.

**Comment by antfu** [comment]
There is an experimental component for that: https://github.com/slidevjs/slidev/blob/main/packages/client/builtin/AutoFitText.vue

Feedback (or better solutions) is greatly welcome :)

**Comment by JPustkuchen** [comment]
Thanks, yes this was more meant as feature request / idea for the future and for discussion. And I think it's different from the AutoFitText as it's for the whole page.

Wasn't meant to be implemented now or by you, just to leave the idea 💡 

Thank you all!

**Comment by antfu** [comment]
There isn't a clear/easy way to do it. I probably won't try it any sooner, but open for PR if someone found a good solution to do so. If you want to keep it as a discussion, I'd suggest you open a discussion instead. Thanks.

**Comment by JPustkuchen** [comment]
Yeah thanks @antfu discussion would have been better, sorry. Is it possible to move it over? (Using GitLab mostly, not very familiar with GitHub). If it's not simple, please don't waste your precious time here, sorry. Thanks for the feedback once again! :)

**Reply by antfu** [reply]
Sure, done :)
